### DISCUSSION TITLE
how to clone sd to nvme?

### DISCUSSION THREAD

**Discussion by unevoljitelj** [Joshua-Riek/ubuntu-rockchip#66] [General]
thank you my dude, this is best working distro for opi5 so far. armbian is nice but hardware accel is pain to setup and i gave it a try few times but failed and never did it 100%. only downside is gnome but ill adapt :) 
i setup your img on sdcard and its working fine but i want it on nvme so i tried cat /dev/mmcblk1 > /dev/nvme0n1 and got permision denied, even with sudo, then tried sudo dd if=/dev/mmcblk1 of=/dev/nvme0n1 and this cloned it but after restart it wont boot from nvme. is there a trick or i am just doing something wrong. these both commands work fine with armbian tho.
thanks for help

**Comment by Joshua-Riek** [comment]
Make sure to flash the bootloader to the SPI with the below commands:
```
sudo dd if=/usr/share/orangepi/rkspi_loader.img of=/dev/mtdblock0 conv=notrunc
sync
```
Then power off your Pi5 and unplug your SD Card. It should boot from the NVMe.

**Comment by Joshua-Riek** [comment]
You could try a fresh install onto the NVMe to see if something is wrong with the cloning process. The below command will write the OS Image to the NVMe, assuming the image is downloaded on your pi.
```
xz -dc ubuntu-22.04-preinstalled-desktop-arm64-orange-pi5.img.xz | sudo dd of=/dev/nvme0n1 bs=4k
```

**Reply by unevoljitelj** [reply]
this worked, thank you

**Comment by mglagolev** [comment]
@Joshua-Riek, thank you very much for your amazing work! I also had no luck using the standard algorithm, but the fresh install worked.

**Comment by stallion25** [comment]
Here is great tutorial, never had problem with it

https://jamesachambers.com/orange-pi-5-ssd-boot-guide/

**Comment by RevBaker** [comment]
I just ran Balena Etcher from the booted SD card and specified CLONE to the NVMe. 

You`ll have to run GPARTED **_before_** you reboot to the NVMe to then resize the operating partition, NOT the boot partition, to the full size (assuming that is what you want to do).
